traits/observers/_list_change_event.py

```diff
@@ -0,0 +1,72 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" Event object for representing mutations to a list.
+"""
+
+# ListChangeEvent is exposed in the public API
+
+
+class ListChangeEvent:
+    """ Event object to represent mutations to a list.
+
+    Attributes
+    ----------
+    trait_list : list
+        The list being mutated.
+    index : int or slice
+        The index used for the mutation.
+    added : list
+        Values added to the list.
+    removed : list
+        Values removed from the list.
+    """
+
+    def __init__(self, *, trait_list, index, removed, added):
+        self.trait_list = trait_list
+        self.added = added
+        self.removed = removed
+        self.index = index
+
+    def __repr__(self):
+        return (
+            "<ListChangeEvent("
```

Comment:
I usually format in `self.__class__.__name__` instead of hard-coding the name, so the `repr` continues to be accurate if someone subclasses the class.

Reply:
Good point. Will do that.